Add test for `dch` capability

eugulixes opened the pull request.

eugulixes committed:
Add test for `dch` capability

dshil reviewed (commented):
I want to see more general test cases.

General corner cases:
- delete 0 characters.
- delete 1 characters.
- delete the whole line.

More complex test cases:
- delete a big number of characters (> terminal rows number)

eugulixes committed:
Tests: add `_check_cap_dch`

eugulixes committed:
Tests: check removing zero characters via `dch`

eugulixes commented:
@dshil, it's better to check removing a character in `dch1` and removing the whole line in `dl1`.

I updated the PR. It's waiting for you review.

dshil commented on gits/test/capabilities_test.py:
```diff
@@ -300,12 +300,15 @@ def test_cap_cup(self):
+        greeting = 'Hello, World!'
+
+        self._check_cap_dch(greeting, 7)  # remove 'Hello, '
+
```
Can we omit this blank line?

dshil commented:
@eugulixes, I agree that it seems more correctly to check deleting the whole line or a character in other tests, but this capability (`cap_dch`) allows us to do the following things:
- Delete a character (not tested).
- Delete a line (not tested).
- Delete a random number of characters (tested).

I think that these cases should be tested in `test_cap_dch`.

eugulixes commented:
@dshil, ok, `dch` and `dl1` delete lines in different ways. But what about removing a character? You suggest testing the same code in `dch` and `dch1`.

dshil commented:
@eugulixes, exactly, because both `dch` and `dch1` allow to do the same thing:
```python
self._cap_dch(1) # delete a character
```

```
self._cap_dch1() # delete a character
```

We assume that `cap_dch1` uses `cad_dch(1)` under the hood, but let's consider these capabilities as separate and test them in the corresponding way.

Let's imagine that `cap_dch` doesn't know anything about `cap_dch1` and will not rely on the fact that deleting a character will be tested in a test for a `cap_dch1` capability.

eugulixes commented:
@dshil, yeah, but in real life we do share code and `dch` and `dch1` are different sides of the same coin. It would be a complete disaster if they were implemented in different ways not knowing about each other. So, let's not imagine this situation=)

eugulixes committed:
Tests: check removing one character via `dch`

eugulixes committed:
Tests: remove `test_cap_dch1`

It's been decided to check all the possible cases in the `dch` capability test.

dshil reviewed (approved):
LGTM

dshil commented:
ping @eugulixes, the PR is ok, but we have some problem, see #55.
Let's fix it in another PR.

eugulixes commented:
@dshil, ok.
